accepted/2024/package-update-command.md

```diff
@@ -0,0 +1,286 @@
+
+```CLI
+C:\ContosoApp\> dotnet package update --vulnerable
+
+Fixing vulnernable packages in ContosoUniversity.sln
+ ContosoLibrary:
+  Upgrading UmbracoForms 8.4.1 to 8.7.1
+
+ ContosoApp:
+  Updated UmbracoForms 8.4.0 to 8.7.1
+    
+Fixed 2 packages in 36 scanned packages.
+```
+
+Using the `--vulnerable` option should update the package to the lowest version that is higher than the currently reference version, which does not have a known vulnerability.
+For example, if `Contoso.Security` version 1.0.0 is currently referenced by the project, and the package source has versions 1.0.1 and 2.0.0 available, 1.0.1 should be selected.
+This is in contrast to `dotnet package update Contoso.Security` which will update to the highest version of the package.
+
+If package identifiers are provided in addition to the `--vulnerable` option, only those packages should be updated.
+Any package provided that does not have a known vulnerability will be skipped.
+
+The `--mode` option will allow customers to choose what strategy of resolving vulnerabilities will be used.
+The `direct` mode would only look for direct package references with known vulnerabilities and upgrade them.
+The `promote` mode would also check transitive packages (equivalent to restore's `<NuGetAuditMode>all`), and add a PackageReference to the package (therefore promoting it to a direct reference).
+Later, more advanced modes can be added.
+When a project is using CPM and transitive pinning is enabled, no PackageReference should be added, just the PackageVersion in the Directory.Packages.props file.
+
+In the future a similar option to fix deprecated versions can be added.
+
+#### --format json
+
+Priority: P2
+
+A JSON schema will be decided closer to implementation.
+A new design document will be created specifically for it, to avoid slowing down the approval of this design document and implementation of other features.
+
+There is potential complexity with designing the schema, as the text output can be tailored to the command line arguments provided.
+For example, `dotnet package update` to update all packages might update just the packages that were updated.
+But `dotnet package update pkg1 pkg2 pkg3` might say pkg1 and pkg2 were updated, but pkg3 was already up to date.
+Similarly, using `--vulnerable` could have vulnerability specific output.
+
+Some consideration may be needed to decide how much of the same information should be reported in the json (particularly when packages are not updated), and if the json schema should be the same for all commands, or if `--vulnerable` should have a different output schema than without `--vulnerable`.
+
+Please provide feedback at <https://github.com/NuGet/Home/issues/14360>
+
+#### Exit Codes
+
+Priority: P1
+
+- 0 - The command will exit with a 0 exit code if at least one package was updated without any error.
+- 1 - The command will exit with a 1 exit code if no packages were updated, but otherwise no error occurred.
```

Comment:
No packages were updated, because we couldn't find appropriate solution or because there were not any updates available?

Basically, what if I run `dotnet package update` on a project with no packages to update?